core/jvm/src/main/scala/scalaz/zio/internal/impls/EnvJVM.scala

```diff
@@ -0,0 +1,133 @@
+package scalaz.zio.internal.impls
+
+import java.util
+import java.util.concurrent._
+
+import scalaz.zio.Exit.Cause
+import scalaz.zio.IO
+import scalaz.zio.internal.Executor.{ Role, Unyielding, Yielding }
+import scalaz.zio.internal.{ Env, ExecutionMetrics, Executor, NamedThreadFactory, Scheduler }
+
+object EnvJVM {
```

Comment:
Does this really need a suffix if it is in `jvm/src/main` already?

Reply:
`currentTime` has been restored and placed in /jvm

Reply:
I don't disagree, but like I said, it's to avoid name conflict with the original `Env` companion object that contains a global counter (when we import the `Env` trait for each impl). We could rename the import or reference the trait by its full package name but that might be even more awkward. What do you think?

Reply:
We can create a `FiberCounter` containing just the `fiberCounter`. Will that work?

Reply:
Yep, I've created the `FiberCounter` object, consolidated the `Env` classes and `RTS` and rebased. 

However there is an unrelated intermittent test failure in `catzSpec` breaking this and other builds:
```
ConcurrentEffect[Task].concurrentEffect.toIO stack safety *** FAILED ***
```